DEVELOPING_OPAL/tools/src/main/scala/org/opalj/support/info/Purity.scala

```diff
@@ -78,45 +101,178 @@ import org.opalj.tac.fpcf.analyses.purity.L1PurityAnalysis
 import org.opalj.tac.fpcf.analyses.purity.L2PurityAnalysis
 import org.opalj.tac.fpcf.analyses.purity.LazyL1PurityAnalysis
 import org.opalj.tac.fpcf.analyses.purity.LazyL2PurityAnalysis
-import org.opalj.tac.fpcf.analyses.purity.SystemOutLoggingAllExceptionRater
 import org.opalj.util.PerformanceEvaluation.time
 import org.opalj.util.Seconds
 
+import org.rogach.scallop.ScallopConf
+import org.rogach.scallop.Subcommand
+
+/**
+ * `PurityConf` is a configuration class for parsing and managing command-line arguments related to purity analysis
+ * in the OPAL framework. It extends `ScallopConf` and mixes in `OpalConf`, providing structured options for various
+ * analysis settings, such as paths, analysis schedulers, and specific analysis configurations.
+ */
+
+class PurityConf(args: Array[String]) extends ScallopConf(args) with OpalConf {
+
+    private object analysis extends Subcommand(AnalysisCommand.name) {
```

Comment:
This is a lot of boiler plate. Can we get make something general out of this that we then just invoke from here?